src/atoms/Heading/Heading.web.js

```diff
@@ -1,5 +1,98 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { getColorKeys, getColor } from '../../_helpers/theme';
+import baseTheme from '../../tokens/theme';
+import automation from '../../_helpers/automation-attributes';
 
-const Heading = () => <div>WIP</div>;
+import Text from '../Text';
+
+const styles = {
+  fontSize({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.size.medium;
+      case 'large':
+        return theme.fonts.size.large;
+      case 'xlarge':
+        return theme.fonts.size.xxlarge;
+      case 'xxlarge':
+        return theme.fonts.size.xxxlarge;
+      case 'xxxlarge':
+        return theme.fonts.size.xxxxlarge;
+      default:
+        return theme.fonts.size.xxxxlarge;
+    }
+  },
+  lineHeight({ theme, size, _weight }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.lineHeight.medium;
+      case 'large':
+        if (theme.fonts.weight[_weight] === theme.fonts.weight.bold) {
+          return theme.fonts.lineHeight.large;
+        }
+        return theme.fonts.lineHeight.medium;
+      case 'xlarge':
+        return theme.fonts.lineHeight.xlarge;
+      case 'xxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeading = styled(Text)`
+  &&& {
+    font-family: ${({ theme, _weight }) => theme.fonts.family.lato[_weight]};
+    font-size: ${styles.fontSize};
+    font-weight: ${({ theme, _weight }) => theme.fonts.weight[_weight]};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
+    line-height: ${styles.lineHeight};
+  }
+`;
+
+const headingLevel = {
+  xxxlarge: 'h1',
+  xxlarge: 'h2',
+  xlarge: 'h3',
+  large: 'h4',
+  medium: 'h5',
+};
+
+const Heading = ({ size, color, children, maxLines, weight }) => {
+  return (
+    <StyledHeading
+      as={headingLevel[size]}
+      size={size}
+      color={color}
+      maxLines={maxLines}
+      _weight={weight}
```

Comment:
Why adding `_`? It's a public prop anyway.

Reply:
Text has `_weight` prop.

Reply:
Okay, got it.